requestmanager/executor/executor.go

```diff
@@ -203,46 +193,17 @@ func (re *requestExecutor) onNewBlockWithPause(block graphsync.BlockData) error
 	return err
 }
 
-func (re *requestExecutor) onNewBlock(block graphsync.BlockData) error {
-	re.doNotSendCids.Add(block.Link().(cidlink.Link).Cid)
-	return re.runBlockHooks(block)
-}
-
-func (re *requestExecutor) processResult(traverser ipldutil.Traverser, link ipld.Link, result types.AsyncLoadResult) error {
-	if result.Err != nil {
-		select {
-		case <-re.ctx.Done():
-			return ipldutil.ContextCancelError{}
-		case re.inProgressErr <- result.Err:
-			traverser.Error(traversal.SkipMe{})
-			return nil
+func (e *Executor) startRemoteRequest(rt RequestTask) error {
+	request := rt.Request
+	if rt.DoNotSendCids.Len() > 0 {
+		cidsData, err := cidset.EncodeCidSet(rt.DoNotSendCids)
+		if err != nil {
+			return err
 		}
+		request = rt.Request.ReplaceExtensions([]graphsync.ExtensionData{{Name: graphsync.ExtensionDoNotSendCIDs, Data: cidsData}})
```

Comment:
does this extension need to be tightly interlinked in this logic, rather than abstracted to a general 'extension' interface?

Reply:
yes cause graphsync directly manages and updates it for pause/resume. BUT, it will still accept the version that is passed in with the call to graphsync.